kinto/core/resource/__init__.py

```diff
@@ -245,6 +245,12 @@ def collection_get(self):
         self._raise_412_if_modified()
 
         headers = self.request.response.headers
+
+        if getattr(self.context, "forced_empty_list", False):
+            # Return an empty list.
+            headers['Total-Records'] = encode_header('0')
+            return self.postprocess([])
+
```

Comment:
Yes, but it seems that in order to save a query we introduce some additional indirection :(